lib/kernel/test/zlib_SUITE.erl

```diff
@@ -1186,6 +1187,17 @@ otp_9981(Config) when is_list(Config) ->
     Ports = lists:sort(erlang:ports()),
     ok.
 
+%% ERIERL-994: the Adler32 checksum returned by the dictionary functionality
+%% could be negative due to returning a signed instead of unsigned integer.
+checksum(Config) when is_list(Config) ->
+    Z = zlib:open(),
+    ok = zlib:deflateInit(Z),
+    TestVec = list_to_binary(lists:duplicate(16, 255)),
+    Chk = zlib:deflateSetDictionary(Z, TestVec),
+    true = Chk >= 0,
+    zlib:close(Z),
+    ok.
```

Comment:
I get a negative Chk with OTP-24. Do you intend to backport this to maint-25/24 or should I do that locally?

Reply:
Fair enough, I'll rebase and include it in the 24- and 25-tracks. :)